cmip6_downscaling/methods/bcsd.py

```diff
@@ -112,15 +112,21 @@ def preprocess_bcsd(
         obs_ds = open_era5(variable, start_year=train_period_start, end_year=train_period_end)
 
         gcm_one_slice = load_cmip(return_type='xr', variable_ids=[variable]).isel(time=0)
-        # calculate and write out the coarsened version of obs dataset to match the gcm
-        # (will be used in training)
-        coarse_obs = regrid_dataset(
-            obs_ds, gcm_one_slice, variable='tasmax', connection_string=connection_string
+        # # calculate and write out the coarsened version of obs dataset to match the gcm
+        # # (will be used in training)
+        coarse_obs, fine_obs_rechunked_path = regrid_dataset(
```

Comment:
it seems confusing that a function called `regrid_dataset` would return both a regridded and a rechunked version of the input dataset. and one of the output is a ds, while the other is a path. maybe we can be a little more compute intensive and break this into two functions.